### PR TITLE
Add UnboundedReaderMaxReadTimeMs to DataflowPipelineDebugOptions, deprecate UnboundedReaderMaxReadTimeSec

### PR DESCRIPTION
**Pull request by stankiewicz**
UnboundedReader is controlled with UnboundedReaderMaxReadTimeSec and UnboundedReaderMaxElements.
For sub second latency user has to set UnboundedReaderMaxElements to 1 which is not flexible as there may be multiple situation where there are more elements.

In this change UnboundedReaderMaxReadTimeSec becomes deprecated. 
UnboundedReaderMaxReadTimeMs is introduced. 
Change is backward compatible - it takes into account both values - If UnboundedReaderMaxReadTimeSec is non default it will take overwrite  UnboundedReaderMaxReadTimeMs.

### CODE REVIEW

**Comment by stankiewicz** [comment]
related to #31082

**Comment by stankiewicz** [comment]
macos-latest

**Comment by stankiewicz** [comment]
run Java Unit Tests (macos-latest)

**Comment by stankiewicz** [comment]
Run Java Unit Tests

**Comment by stankiewicz** [comment]
R: @jrmccluskey

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/options/DataflowPipelineDebugOptions.java [hunk 1]
```diff
@@ -216,14 +216,27 @@ public Dataflow create(PipelineOptions options) {
...
+  /**
+   * The max amount of time an UnboundedReader is consumed before checkpointing.
+   *
+   * @deprecated use {@link DataflowPipelineDebugOptions#getUnboundedReaderMaxReadTimeMs()} instead
+   */
   @Description(
       "The max amount of time before an UnboundedReader is consumed before checkpointing, in seconds.")
   @Default.Integer(10)
+  @Deprecated
   Integer getUnboundedReaderMaxReadTimeSec();
```
Comment (scwhittle) [inline runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/options/DataflowPipelineDebugOptions.java:228]: what about instead changing this to a Long?

**Review by scwhittle — CHANGES_REQUESTED** [review]

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/options/DataflowPipelineDebugOptions.java (hunk 1 above)
Comment (scwhittle) [reply runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/options/DataflowPipelineDebugOptions.java:228]: oh, I see the related issue now that type was incompatible. looking into that

**Review by scwhittle — CHANGES_REQUESTED** [review]

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/options/DataflowPipelineDebugOptions.java [hunk 2]
```diff
@@ -216,14 +216,42 @@ public Dataflow create(PipelineOptions options) {
...
-  /** The max amount of time an UnboundedReader is consumed before checkpointing. */
+  /**
+   * The max amount of time an UnboundedReader is consumed before checkpointing.
+   *
+   * @deprecated use {@link DataflowPipelineDebugOptions#getUnboundedReaderMaxReadTimeMs()} instead
+   */
   @Description(
       "The max amount of time before an UnboundedReader is consumed before checkpointing, in seconds.")
-  @Default.Integer(10)
+  @Deprecated
```
Comment (scwhittle) [inline runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/options/DataflowPipelineDebugOptions.java:226]: nit: does keeping the Default.Integer(10) here work? then you could remove the null check and default from the factory?

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/options/DataflowPipelineDebugOptions.java [hunk 3]
```diff
@@ -216,14 +216,42 @@ public Dataflow create(PipelineOptions options) {
...
+  @Default.InstanceFactory(UnboundedReaderMaxReadTimeFactory.class)
+  Integer getUnboundedReaderMaxReadTimeMs();
+
+  void setUnboundedReaderMaxReadTimeMs(Integer value);
+
+  /**
+   * Sets Integer value based on old, deprecated field ({@link
+   * DataflowPipelineDebugOptions#getUnboundedReaderMaxReadTimeSec()}) is set.
+   */
+  class UnboundedReaderMaxReadTimeFactory implements DefaultValueFactory<Integer> {
```
Comment (scwhittle) [inline runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/options/DataflowPipelineDebugOptions.java:243]: can this be static final class?

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/options/DataflowPipelineDebugOptions.java [hunk 4]
```diff
@@ -216,14 +216,42 @@ public Dataflow create(PipelineOptions options) {
...
+  @Description(
+      "The max amount of time before an UnboundedReader is consumed before checkpointing, in millis.")
+  @Default.InstanceFactory(UnboundedReaderMaxReadTimeFactory.class)
+  Integer getUnboundedReaderMaxReadTimeMs();
+
+  void setUnboundedReaderMaxReadTimeMs(Integer value);
+
+  /**
+   * Sets Integer value based on old, deprecated field ({@link
+   * DataflowPipelineDebugOptions#getUnboundedReaderMaxReadTimeSec()}) is set.
```
Comment (scwhittle) [inline runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/options/DataflowPipelineDebugOptions.java:241]: rm "is set"

**Review by scwhittle — CHANGES_REQUESTED** [review]

**Pull request merged**